src/XRP/model/xrp-issued-currency.ts

```diff
@@ -19,20 +21,32 @@ export default class XrpIssuedCurrency {
     issuedCurrency: IssuedCurrencyAmount,
   ): XrpIssuedCurrency | undefined {
     const currency = issuedCurrency.getCurrency()
-    const xrpCurrency = currency && XrpCurrency.from(currency)
+    if (!currency) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Issued currency protobuf does not contain valid `currency` field.',
+      )
+    }
+    const xrpCurrency = XrpCurrency.from(currency)
 
     let value
     try {
       value = bigInt(issuedCurrency.getValue())
     } catch {
-      value = undefined
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Issued currency protobuf does not contain valid `value` field.',
+      )
     }
 
-    if (value) {
-      const issuer = issuedCurrency.getIssuer()?.getAddress()
-      return new XrpIssuedCurrency(xrpCurrency, value, issuer)
+    const issuer = issuedCurrency.getIssuer()?.getAddress()
+    if (!issuer) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Issued currency protobuf does not contain valid `issuer` field.',
```

Comment:
```suggestion
        'IssuedCurrency protobuf does not contain `issuer` field.',
```

Reply:
in this case, that if statement is only going to be triggered if `issuer` is undefined, so I feel like it makes more sense to be specific in our error message that the field isn't merely invalid, but it doesn't even exist.